variables/Four Column Regex.tpl

```diff
@@ -0,0 +1,342 @@
+          {
+            "paramName": "convertTrue",
+            "type": "EQUALS",
+            "paramValue": true
+          }
+        ],
+        "displayName": "",
+        "simpleValueType": true,
+        "name": "convertTrueTo",
+        "type": "TEXT"
+      },
+      {
+        "simpleValueType": true,
+        "name": "convertFalse",
+        "checkboxText": "Convert false to…",
+        "type": "CHECKBOX"
+      },
+      {
+        "enablingConditions": [
+          {
+            "paramName": "convertFalse",
+            "type": "EQUALS",
+            "paramValue": true
+          }
+        ],
+        "displayName": "",
+        "simpleValueType": true,
+        "name": "convertFalseTo",
+        "type": "TEXT"
+      }
+    ]
+  }
+]
+
+
+___SANDBOXED_JS_FOR_WEB_TEMPLATE___
+
+const makeString = require('makeString');
+const inputVar = makeString(data.inputVar);
+const table = data.regexTable;
+const defaultValue = data.defaultValue;
+//Advanced Settings Values
+const ignoreCase = data.ignoreCase;
+const fullMatch = data.fullMatch;
+const capRep = data.captureReplace;
+//Format Value Values
+const convertCaseTo = data.convertCaseTo;
+const convertCase = data.convertCase;
+const convertNullTo = data.convertNullTo;
+const convertNull = data.convertNull;
+const convertUndefinedTo = data.convertUndefinedTo;
+const convertUndefined = data.convertUndefined;
+const convertTrueTo = data.convertTrueTo;
+const convertTrue = data.convertTrue;
+const convertFalseTo = data.convertFalseTo;
+const convertFalse = data.convertFalse;
+//Declare Variables
+  //If you need to change your default delimter, do it here.
+var val = defaultValue !== undefined ? defaultValue.split('|') : defaultValue,
+    a,
+    b;
+if(typeof(table) !=='undefined'){
+  //Itterate through the table
+  for(a=0;a<table.length;a++){
+    if(ignoreCase===true){
+      //Convert regex and input to lowercase
+      var input = inputVar.toLowerCase(),
+          re = table[a].inputVal.toLowerCase();
+    } else{
+      var input = inputVar,
+          re = table[a].inputVal;
+    }
+    if(input.match(re) !== null){
+      if (fullMatch===true){
+        if(input.match(re)[0]==input){
+          val=[table[a].output1,table[a].output2,table[a].output3];
+          if (capRep === true){
+            val = [table[a].output1,table[a].output2,table[a].output3];
+            for(b=0;b<val.length;b++){
+              if(val[b].match('\\$\\d+') !== null){
+                val[b] = input.match(re)[val[b].match('\\d+')[0]];
+              }
+            }
+          }
+          break;
+        } else {
+          continue;
+        }
+      } else {
+        val = [table[a].output1,table[a].output2,table[a].output3];
+        if (capRep === true){
+          for(a=b;b<val.length;b++){
+            if(val[b].match('\\$\\d+') !== null){
+              val[b] = input.match(re)[val[b].match('\\d+')[0]];
+            }
+          }
+        }
+        break;
+      }
+    }
+  }
+}
```

Comment:
That's a LOT of nesting... also, the two blocks (if and else) are almost identical - just the `a=b` is different. Sounds like they could be converged.